### PR TITLE
Improvement of props script

### PR DESCRIPTION
**Pull request by Getriax**
**Fixes and changes:**

- Now bob will notify a user if there is no PROPS_API_TOKEN available, (@krzyurb idea 👋)
- Bob will also notify if user tries to props h[im/er]self.
- Test cases for these situations

### CODE REVIEW

File: src/props/props.controller.js [hunk 1]
```diff
@@ -9,14 +9,27 @@ class PropsController {
     this.logger = logger;
   }
 
+  propsApiTokenIsNotAvailable() {
+    const { PROPS_API_TOKEN } = process.env;
+    return PROPS_API_TOKEN === '' || PROPS_API_TOKEN === undefined;
+  }
+
   async giveProps(res) {
+    if (this.propsApiTokenIsNotAvailable()) {
```
Comment (krzyurb) [inline src/props/props.controller.js]: Why not simply `if (!process.env.PROPS_API_TOKEN)` ?

File: src/props/props.controller.js [hunk 2]
```diff
@@ -9,14 +9,27 @@ class PropsController {
     this.logger = logger;
   }
 
+  propsApiTokenIsNotAvailable() {
+    const { PROPS_API_TOKEN } = process.env;
+    return PROPS_API_TOKEN === '' || PROPS_API_TOKEN === undefined;
```
Comment (Davlis) [inline src/props/props.controller.js]: Consider changing to
```javascript
return !PROPS_API_TOKEN;
```

File: test/props.spec.js [hunk 3]
```diff
@@ -31,13 +31,28 @@ describe('give_props <users> <text>', () => {
       ]);
 
       sinon.stub(propsAPI, 'giveProp').resolves([]);
+
+      process.env.PROPS_API_TOKEN = 'example';
```
Comment (Davlis) [inline test/props.spec.js]: You can also define this environment in `.env.test.example` and `.env.test`

File: test/props.spec.js [hunk 4]
```diff
@@ -31,13 +31,28 @@ describe('give_props <users> <text>', () => {
...
     });
 
     afterEach(() => {
       sinon.restore();
       room.destroy();
     });
 
+    it('Should notify a user about missing api token', async () => {
+      process.env.PROPS_API_TOKEN = '';
```
Comment (Davlis) [inline test/props.spec.js]: In my opinion, stubing this is much more cleaner, what do you think?

File: src/props/props.controller.js (hunk 1 above)
Comment (Getriax) [reply src/props/props.controller.js]: I think that a function is more descriptive, that is why I've chosen this way.
Of course, both examples have their pros.

File: src/props/props.controller.js (hunk 2 above)
Comment (Getriax) [reply src/props/props.controller.js]: Done!

File: test/props.spec.js (hunk 4 above)
Comment (Getriax) [reply test/props.spec.js]: Fixed!

File: test/props.spec.js (hunk 3 above)
Comment (Getriax) [reply test/props.spec.js]: Done!

File: test/props.spec.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ describe('give_props <users> <text>', () => {
     let room = null;
     let propsService;
     let propsAPI;
+    console.log('TOKEN', process.env.PROPS_API_TOKEN);
```
[the remark this answers is not in the document]
Comment (Getriax) [reply test/props.spec.js]: Left it by mistake, fixed!

File: src/props/props.controller.js [hunk 6]
```diff
@@ -9,14 +9,27 @@ class PropsController {
     this.logger = logger;
   }
 
+  propsApiTokenIsNotAvailable() {
```
Comment (Davlis) [inline src/props/props.controller.js]: Consider returning `PROPS_API_TOKEN` directly from `process.env` 

```javascript
propsApiTokenIsNotAvailable() {
  return !process.env.PROPS_API_TOKEN;
}
```

Comment (Getriax) [reply src/props/props.controller.js]: Done!

**Pull request merged**